design_drafts/components_architecture_and_urdf_examples.md

```diff
@@ -0,0 +1,424 @@
+      <param name="min_values">-1</param>
+      <param name="max_values">1</param>
+    </joint>
+  </ros2_control>
+```
+
+Note:
+  * `ros2_control_components/PositionJoint`type has implicitly:
+    ```xml
+      <commandInterfaceType>position</commandInterfaceType>
+      <stateInterfaceType>position</stateInterfaceType>
+    ```
+
+#### 2. Industrial Robots with multiple interfaces (can not be written at the same time)
+  * the communication is done using proprietary API to communicate with robot control box
+  * Data for all joints is exchanged in batch (at once)
+  * Examples: KUKA FRI, ABB Yummy, Schunk LWA4p, etc.
+
+```xml
+  <ros2_control name="2DOF_System_Robot_MultiInterface" type="system">
+    <hardware>
+      <classType>ros2_control_demo_hardware/2DOF_System_Hardware_MultiInterface</classType>
+      <param name="example_param_write_for_sec">2</param>
+      <param name="example_param_read_for_sec">2</param>
+    </hardware>
+    <joint name="joint1">
+      <classType>ros2_control_components/MultiInterfaceJoint</classType>
+      <commandInterfaceType>position</commandInterfaceType>
+      <commandInterfaceType>velocity</commandInterfaceType>
+      <commandInterfaceType>effort</commandInterfaceType>
+      <stateInterfaceType>position</stateInterfaceType>
+      <stateInterfaceType>velocity</stateInterfaceType>
+      <stateInterfaceType>effort</stateInterfaceType>
+      <param name="min_position_value">-1</param>
+      <param name="max_position_value">1</param>
+      <param name="min_velocity_value">-1</param>
+      <param name="max_velocity_value">1</param>
+      <param name="min_effort_value">-0.5</param>
+      <param name="max_effort_value">0.5</param>
+    </joint>
+    <joint name="joint2">
+      <classType>ros2_control_components/MultiInterfaceJoint</classType>
+      <commandInterfaceType>position</commandInterfaceType>
+      <stateInterfaceType>position</stateInterfaceType>
+      <stateInterfaceType>velocity</stateInterfaceType>
+      <stateInterfaceType>effort</stateInterfaceType>
+      <param name="min_position_value">-1</param>
+      <param name="max_position_value">1</param>
+      <param name="min_velocity_value">-1</param>
+      <param name="max_velocity_value">1</param>
+      <param name="min_effort_value">-0.5</param>
+      <param name="max_effort_value">0.5</param>
+    </joint>
+  </ros2_control>
+```
+
+Note:
+  * For `joint2` the "velocity" and "effort" command interfaces are intentionally left out to show another common use-case
+
+
+#### 3. Robots with multiple interfaces used at the same time
+  * the communication is done using proprietary API or some standardized interface
+  * Data for all joints is exchanged in batch (at once)
+  * Examples: (some humanoid robots)?
+
+```xml
+  <ros2_control name="2DOF_System_Robot_MultiInterface_MultiWrite" type="system">
+    <hardware>
+      <classType>ros2_control_demo_hardware/2DOF_System_Hardware_MultiInterface_MultiWrite</classType>
```

Comment:
consider leaving out this use case

Reply:
You are right, there is no really a value in it. I will remove it but leave the use-case named. This will be cleaned when finalizing the document.